Rakefile

```diff
@@ -38,67 +39,110 @@ RSpec::Core::RakeTask.new(:coverage) do |t|
   t.rcov_opts = ['--exclude', 'spec']
 end
 
+default_reference_platform = 'windows2012r2-64'
+default_acceptance_platform = 'centos7-64mdca-windows2012r2-64a'
+internal_pe_host_location = 'http://neptune.puppetlabs.lan/'
+internal_pe_version = '2016.5/ci-ready/'
 
-platform = ENV["PLATFORM"]
+# Modify a rototiller command and add the common testing options
+def add_common_beaker_options_to(command)
+  command.add_option do |debug|
+    debug.name = '--debug'
+  end
+  command.add_option do |preserve_hosts|
+    preserve_hosts.name = '--preserve-hosts'
+    preserve_hosts.add_argument do |arg|
+      arg.name = 'never'
+      arg.add_env({:name => 'BEAKER_PRESERVEHOSTS', :message => 'Override the --preserve-hosts option'})
+    end
+  end
+  command.add_option do |config|
+    config.name = '--config'
+    config.add_argument do |arg|
+        arg.name = "tests/configs/#{ENV['PLATFORM']}"
+      end
+  end
+  command.add_option do |keyfile|
+    keyfile.name = '--keyfile'
+    keyfile.add_argument do |arg|
+      arg.name = "#{ENV['HOME']}/.ssh/id_rsa-acceptance"
+      arg.add_env({:name => 'BEAKER_KEYFILE', :message => 'Override the --keyfile option'})
+    end
+  end
+  command.add_option do |loadpath|
+    loadpath.name = '--load-path'
+    loadpath.add_argument do |arg|
+      arg.name = 'tests/lib'
+      arg.add_env({:name => 'BEAKER_LOADPATH', :message => 'Override the --load-path option' })
+    end
+  end
+end
 
-# Create the directory, if it exists already you'll get an error, but this should not stop the execution
-begin
-  sh 'mkdir tests/configs'
-rescue => e
-  puts e.message
+desc 'Generate Beaker Host config'
+rototiller_task :host_config, [:default_platform] do |t, args|
+  t.add_env({:name => 'PLATFORM', :message => 'PLATFORM Must be set. For example "windows2012r2-64"', :default => args[:default_platform]})
+  hosts_file = "tests/configs/#{ENV['PLATFORM']}"
+  t.add_command do |cmd|
+    cmd.name = 'bundle exec beaker-hostgenerator'
+    cmd.add_argument({:name => "#{ENV['PLATFORM']} > #{hosts_file}"})
+  end
 end
 
+# Runs the reference tests in agent only configuration
 desc 'Executes reference tests (agent only) intended for use in CI'
-task :reference_tests do
-  command = "bundle exec beaker-hostgenerator --global-config {masterless=true} #{platform} > tests/configs/#{platform}" # should we assume the "configs" directory is present?
-  sh command
-
-  command =<<-EOS
-bundle exec beaker                          \
-    --debug                                 \
-    --preserve-hosts never                  \
-    --config tests/configs/$PLATFORM        \
-    --keyfile ~/.ssh/id_rsa-acceptance      \
-    --load-path tests/lib                   \
-    --type aio                              \
-    --pre-suite tests/reference/pre-suite   \
-    --tests tests/reference/tests
-    EOS
-  sh command
+rototiller_task :reference_tests  do |task|
```

Comment:
Same here, since this gem won't be installed on Windows and we still want rake to work for other tasks.